ext/filter/filter.stub.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+function filter_has_var(int $type, string $variable_name): bool {}
+
+/**
+ * @param mixed $options
+ * @return mixed
+ */
+function filter_input(int $type, string $variable_name, int $filter = FILTER_DEFAULT, $options = NULL) {}
+
+/**
+ * @param mixed $variable
+ * @param mixed $options
+ * @return mixed
+ */
+function filter_var($variable, int $filter = FILTER_DEFAULT, $options = NULL) {}
+
+/**
+ * @param mixed $options
+ * @return mixed
```

Comment:
I think that here and for `filter_var_array` the return type should be `array|false|null`.

Reply:
it's right return 3 types? i didn't see any other case.

seeing in `filter.c - line 747` the docblock specify that results a mixed return.

Reply:
It's a bit hard to tell for sure -- let's just leave it at mixed for now, it's not like it makes a difference right now :)